transport/http/handler.go

```diff
@@ -62,13 +63,48 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	service := popHeader(req.Header, ServiceHeader)
 	procedure := popHeader(req.Header, ProcedureHeader)
 	bothResponseError := popHeader(req.Header, AcceptsBothResponseErrorHeader) == AcceptTrue
-	// add response header to echo accepted rpc-service
+
+	// Extract tracing context and remove tracing headers before they reach handlers
+	carrier := opentracing.HTTPHeadersCarrier(req.Header)
+	parentSpanCtx, _ := h.tracer.Extract(opentracing.HTTPHeaders, carrier)
+
+	// Remove tracing headers so application handlers don't see them
+	for k := range req.Header {
+		kLower := strings.ToLower(k)
+		if kLower == strings.ToLower(UberTraceIDHeader) || strings.HasPrefix(kLower, strings.ToLower(UberCtxHeader)) {
+			req.Header.Del(k)
+		}
+	}
+
+	// Start span
+	start := time.Now()
+	tags := opentracing.Tags{
+		"rpc.service":   service,
+		"rpc.transport": "http",
```

Comment:
1. We have constants TransportName and TransportHTTP2Name here.
2. It could be http2, please see TransportHTTP2Name usage examples.